lib/Controller/UsersController.php

```diff
@@ -469,6 +501,254 @@ public function create($username, $password, array $groups= [], $email='') {
 		);
 	}
 
+	/**
+	 * Set password for user using link
+	 *
+	 * @PublicPage
+	 * @NoCSRFRequired
+	 * @NoAdminRequired
+	 * @NoSubadminRequired
+	 *
+	 * @param string $token
+	 * @param string $userId
+	 * @return TemplateResponse
+	 */
+	public function setPasswordForm($token, $userId) {
+		try {
+			$this->checkPasswordSetToken($token, $userId);
+		} catch (UserTokenException $e) {
+			if ($e->getCode() === UserTokenException::TOKEN_TIME_EXPIRED) {
+				return new TemplateResponse(
+					'user_management', 'new_user/resendtokenbymail',
+					[
+						'link' => $this->urlGenerator->linkToRouteAbsolute('user_management.Users.resendToken', ['userId' => $userId])
+					], 'guest'
+				);
+			}
+			$this->log->logException($e, ['app' => 'user_management']);
+			$tokenMessage = null;
+			if ($e->getCode() === UserTokenException::TOKEN_AND_TIME_INVALID) {
+				$tokenMessage = $this->l10n->t('The token provided is invalid.');
+			} else {
+				$tokenMessage = $this->l10n->t('The token mismatch happened.');
```

Comment:
"The token does not match"

Reply:
or why do we need a different message than "The provided token is invalid" ? if the token doesn't match it's invalid anyway ?

Reply:
For token mismatch and token invalid will have -> `The token provided is invalid.`